lib/HLSL/DxilValidation.cpp

```diff
@@ -3994,9 +3994,7 @@ static void ValidateWaveSize(ValidationContext &ValCtx,
                                           Metadata::ConstantAsMetadataKind;
       DXASSERT(!invalidPropertyTag,
```

Comment:
This should cause a warning/error on an unused variable in Release builds. You'll either need to collapse the condition into the assert or apply `[[maybe_unused]]` to the variable declaration to silence the warning/error/